payments-ui-core/src/main/java/com/stripe/android/ui/core/StripeCardScanProxy.kt

```diff
@@ -0,0 +1,108 @@
+package com.stripe.android.ui.core
+
+import androidx.annotation.IdRes
+import androidx.appcompat.app.AppCompatActivity
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentManager
+import androidx.lifecycle.LifecycleOwner
+import com.stripe.android.BuildConfig
+import com.stripe.android.stripecardscan.cardscan.CardScanSheet
+import com.stripe.android.stripecardscan.cardscan.CardScanSheetResult
+
+/**
+ * Proxy to access stripecardscan code safely
+ *
+ */
+internal interface StripeCardScanProxy {
+    fun present()
+
+    fun attachCardScanFragment(
+        lifecycleOwner: LifecycleOwner,
+        supportFragmentManager: FragmentManager,
+        @IdRes fragmentContainer: Int,
+        onFinished: (cardScanSheetResult: CardScanSheetResult) -> Unit
+    )
+
+    companion object {
+        fun create(
+            fragment: Fragment,
+            stripePublishableKey: String,
+            onFinished: (cardScanSheetResult: CardScanSheetResult) -> Unit,
+            provider: () -> StripeCardScanProxy = {
+                DefaultStripeCardScanProxy(CardScanSheet.create(fragment, stripePublishableKey, onFinished))
+            },
+            isStripeCardScanAvailable: IsStripeCardScanAvailable = DefaultIsStripeCardScanAvailable()
+        ): StripeCardScanProxy {
+            return if (isStripeCardScanAvailable()) {
+                provider()
+            } else {
+                UnsupportedStripeCardScanProxy()
+            }
+        }
+
+        fun create(
+            activity: AppCompatActivity,
```

Comment:
Can this be a `ComponentActivity` like PaymentSheet and PaymentLauncher's APIs use?

Reply:
StripeCardScanProxy proxies the CardScanSheet, whose create function takes an AppCompatActivity, so I think it makes sense for this proxy to do the same